src/root.jl

```diff
@@ -271,52 +296,33 @@ end
 # ┌─────────┬────────────────────────────────┬───┬────────────┬───┐
 # │ TKey    │ content                        │ X │ offsets    │ x │
 # └─────────┴────────────────────────────────┴───┴────────────┴───┘
-#           │←        fLast - fKeylen       →│                    │
+#           │←        fLast - fKeylen       →│(l1)                │
 #           │                                                     │
 #           │←                       fObjlen                     →│
 #
-function readbasketbytes!(data, offsets, io, idx)
-    basketkey = unpack(io, TBasketKey)
+function readbasketbytes!(local_io, basketkey, contentsize, idx)
+    basketrawbytes = read(datastream(local_io, basketkey))
+    Keylen = basketkey.fKeylen
 
-    s = datastream(io, basketkey)  # position(s) == 0, but offsets start at -basketkey.fKeylen
-    start = position(s)
-    contentsize = basketkey.fLast - basketkey.fKeylen
     offsetbytesize = basketkey.fObjlen - contentsize - 8
-    offset_len = offsetbytesize ÷ 4 # these are always Int32
+    offsetnumints = offsetbytesize ÷ 4 # these are always Int32
+    l1 = contentsize + 4 # see the graph above
 
+    data = @view basketrawbytes[begin:contentsize]
```

Comment:
`begin` isn't in Julia v1.0.  You may have to use `firstindex(basketrawbytes)` instead

Reply:
I changed to hard code `1`. I doubt we'd support 1.0 in our 1.0 release, the shared file system is un-usable without the tarball registry, and also pre-1.5 TTFP is exacerbated by poor hardware.